common/cache-t.hpp

```diff
@@ -16,6 +16,7 @@ class GlobalCacheBase
   const char * name;
   GlobalCacheBase * next;
   GlobalCacheBase * * prev;
+  static Mutex global_cache_lock;
 protected:
```

Comment:
Please add a comment to why this needs to be in the class.  By following the downstream bug report I was able to make an educated guess, but I want to make the reason to be documented in the code itself.